api/src/main/java/jakarta/servlet/http/Cookie.java

```diff
@@ -428,4 +441,56 @@ public void setHttpOnly(boolean isHttpOnly) {
     public boolean isHttpOnly() {
         return isHttpOnly;
     }
+    
+    /**
+     * Sets the value of the cookie attribute associated with the given name.
+     * 
+     * <p>
+     * This should override any predefined attribute for which already a getter/setter pair exist in the current version.
+     * E.g. when <code>cookie.setAttribute("domain", domain)</code> is invoked,
+     * then any entry in the mapping returned by {@link #getAttributes()} associated with key "domain", case insensitive,
+     * should take precedence over <code>cookie.getDomain()</code>.
+     * 
+     * @param name the name of the cookie attribute to set the value for, case insensitive
+     * 
+     * @param value the value of the cookie attribute associated with the given name, can be {@code null}
+     *
+     * @throws IllegalArgumentException if the cookie attribute name is null or empty, or if either the cookie attribute name or value
+     * contains any illegal characters (for example, a comma, space, or semicolon) or matches a token reserved for use by the cookie protocol
+     * 
+     * @since Servlet 5.1
+     */
+    public void setAttribute(String name, String value) {
+        if (name == null || name.isEmpty()) {
```

Comment:
The implementation here needs to enforce the apidoc above that says: 
> This should override any predefined attribute for which already a getter/setter pair exist in the current version.

This can be done be removing all the fields and using only the map to store the attributes (and then updating the getters/setters accordingly).  Alternately, known fields can be intercepted here and call the setters directly.

Eitherway, we need for the known fields to always appear in the map or to never appear in the map, and never have different values if they appear in both.

Reply:
That won't work in long term. An good example is the earlier proposal to make `SameSite` an enum. We want the API to behave in best possible compatible way when HTTP guys at some point decide to add a new value to the enumeration.

Reply:
I've prepped another PR for this: https://github.com/eclipse-ee4j/servlet-api/pull/400